include/deal.II/base/table_indices.h

```diff
@@ -66,12 +66,14 @@ class TableIndices
   /**
    * Read-only access the value of the <tt>i</tt>th index.
    */
-  DEAL_II_CONSTEXPR std::size_t operator[](const unsigned int i) const;
+  DEAL_II_CONSTEXPR std::size_t
+                    operator[](const unsigned int i) const;
```

Comment:
These look weird but now that we have C++14 perhaps we can fix it by just promoting these from `DEAL_II_CONSTEXPR` to `constexpr`.

Reply:
We know that at least MSVC has problems and the promotion would not work in general. https://github.com/dealii/dealii/blob/master/cmake/checks/check_01_cxx_features.cmake#L572-L621
Of course, we could try in selected places.

Reply:
I'll give it a try

Reply:
We fixed a lot of these in #11122 but we can't yet get rid of `DEAL_II_CONSTEXPR` - we need it to compile with some older compilers.